src/mo_frontend/typing.ml

```diff
@@ -2300,8 +2302,14 @@ and infer_block_exps env decs : T.typ =
 and infer_dec env dec : T.typ =
   let t =
   match dec.it with
-  | ExpD exp
-  | LetD (_, exp) ->
+  | ExpD exp 
+  | LetD (_, exp, None) -> infer_exp env exp
+  | LetD (_, exp, Some fail) -> 
+    let fail_type = infer_exp env fail in
+    if not T.(sub fail_type Non) then 
```

Comment:
maybe there is a nicer way to put this? @crusso

Reply:
`infer_exp` or `infer_exp_strong`?

Reply:
Seems like check_exp again None type is working